[React-Native] Add FlatList and SectionList to Animated

GertjanReynaert opened the pull request:
Since [React-Native 57](https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md#added-new-features-4) FlatList and SectionList are added to the exports of Animated. This pr reflects that addition in the type definitions.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md#added-new-features-4
- [ ] Increase the version number in the header if appropriate.
  - Not appropriate, since this change is already present in RN57, which is the current version of these types
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.
  - No substantial changes (I guess? 😅 )

Esemesek commented:
@Gebatzens Can you fill template correctly and remove unnecessary sections? Also, can you add tests and link part of `react-native` documentation or source code which approves your changes? https://facebook.github.io/react-native/docs/animated#animatable-components -> Docs are missing those 2 components.

GertjanReynaert commented:
@Esemesek I removed and updated the correct sections. I didn't ran any tests, since I did this edit in the Github UI so I don't have the definitely typed repo cloned locally. However these changes are just a copy of what I did in my `@types/react-native/index.d.ts` in my local project, so this should work.

henrikra commented:
Is there a way to type animated components properly? Because now in JSX you don't get autocompletion for props

iRoachie commented:
Actually wondering why we use `any` as the type here. Shouldn't it use the interface from the actual component?

henrikra commented:
@iRoachie Exactly my thoughts :D
